org.osgi.test.common/src/main/java/org/osgi/test/common/annotation/config/ConfigEntry.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) OSGi Alliance (2020). All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.common.annotation.config;
+
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+@Inherited
+@Target({
+	ANNOTATION_TYPE
+})
+@Retention(RUNTIME)
+@Documented
+public @interface ConfigEntry {
```

Comment:
Just a thought: I've recently started playing with OSGi MetaTypes. Are we re-inventing the wheel here? Could we leverage the metatype syntax to avoid having to create/maintain our own custom syntax for config entries? Or is there some functionality that metatypes doesn't give us? Or do metatypes give us too much functionality that `ConfigAdmin` doesn't support?

This is just a thought bubble - I don't have a lot of experience with OSGi meta types or with `ConfigAdmin`, so I don't have a strong feel for what is appropriate. I just wanted to throw it out there for consideration.

Reply:
i thought that metatype is is just compiletime to create these metatype xml and not reading an annotation. But i think that only shows that i do not know enough about it.

Reply:
You're right - metatype annotations do not have runtime retention and so they couldn't be adapted to this usage.